### ISSUE TITLE
[Bug]: Messages not appearing in UI

### ISSUE THREAD

**Issue by hickey** [hickey/meshchat#15]
### Version

v2.9

### System Type

AREDN node

### What happened?

Installed the latest build (non-release) on a test node. There are no messages displayed. The message database on the node is populated with valid data. 

The output of the javascript console follows:

```
GET
http://172.28.100.200:8080/meshchat/
[HTTP/1.1 200 OK 5ms]

GET
http://172.28.100.200:8080/meshchat/skeleton.css
[HTTP/1.1 200 OK 7ms]

GET
http://172.28.100.200:8080/meshchat/style.css
[HTTP/1.1 200 OK 14ms]

GET
http://172.28.100.200:8080/meshchat/jquery-2.2.0.min.js
[HTTP/1.1 200 OK 31ms]

GET
http://172.28.100.200:8080/meshchat/ohsnap.js
[HTTP/1.1 200 OK 35ms]

GET
http://172.28.100.200:8080/meshchat/js.cookie.js
[HTTP/1.1 200 OK 36ms]

GET
http://172.28.100.200:8080/meshchat/shared.js
[HTTP/1.1 200 OK 39ms]

GET
http://172.28.100.200:8080/meshchat/chat.js
[HTTP/1.1 200 OK 4ms]

Expected declaration but found ‘;’.  Skipped to next declaration. [style.css:306:17](http://172.28.100.200:8080/meshchat/style.css)
GET
http://172.28.100.200:8080/meshchat/skeleton.css
[HTTP/1.1 200 OK 4ms]

GET
http://172.28.100.200:8080/meshchat/style.css
[HTTP/1.1 200 OK 5ms]

Unknown pseudo-class or pseudo-element ‘has’.  Ruleset ignored due to bad selector. [meshchat:1:538](http://172.28.100.200:8080/meshchat/)
Expected declaration but found ‘;’.  Skipped to next declaration. [style.css:306:17](http://172.28.100.200:8080/meshchat/style.css)
XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=config
[HTTP/1.1 200 OK 449ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=messages_version_ui&call_sign=WT0F&id=xxQBI&epoch=1699546583&_=1699546583855
[HTTP/1.1 200 OK 463ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=users&call_sign=WT0F&id=xxQBI&_=1699546583856
[HTTP/1.1 200 OK 460ms]

GET
http://172.28.100.200:8080/meshchat/alert.mp3
[HTTP/1.1 200 OK 4ms]

Use of drawWindow method from CanvasRenderingContext2D is deprecated. Use tabs.captureTab extensions API instead https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/tabs/captureTab [pagedata-script.js:1:12509](moz-extension://4b5ea91b-b84a-6d4b-b000-0463eaa0f382/content/pagedata-script.js)
GET
http://172.28.100.200:8080/favicon.ico
[HTTP/1.1 404 Not Found 1ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=messages&call_sign=WT0F&id=xxQBI&epoch=1699546584&_=1699546583857
[HTTP/1.1 200 OK 243ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=messages_version_ui&call_sign=WT0F&id=xxQBI&epoch=1699546598&_=1699546583858
[HTTP/1.1 200 OK 291ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=users&call_sign=WT0F&id=xxQBI&_=1699546583859
[HTTP/1.1 200 OK 287ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=messages_version_ui&call_sign=WT0F&id=xxQBI&epoch=1699546613&_=1699546583860
[HTTP/1.1 200 OK 297ms]

XHRGET
http://172.28.100.200:8080/cgi-bin/meshchat?action=users&call_sign=WT0F&id=xxQBI&_=1699546583861
```

### What browsers are you seeing the problem on?

Firefox

**Comment by hickey** [comment]
Further investigation finds that the `messages` call is returning an empty JSON document.

**Comment by hickey** [comment]
After extensive debugging, found that `nws-alerts` was creating a bad message DB entry. No call sign was being created on the message entry. This has to do with the `send_message` call not having `epoch` defined. This caused LUA to throw an exception when `query.epoch` was a `nil` object which for some (uncertain) reason caused `query.call_sign` to have and empty or `nil` value. 

Should add to the `send_message` function a check to see if epoch is not defined and if so just to use the current time.

**Comment by n2mh** [comment]
When developing code about a year ago, perhaps longer than that, to push weather alerts into Meshchat, I found that unlike ver 1.x, ver 2.x would not tolerate a space char in the channel field. IIRC, with a space in the char field, the message appeared to be accepted but did not display once ingested into the system. For example, in 1.x

`NWR SAME Alert` 

as a channel name worked but in 2.x it did not. I ultimately went to `NWR-SAME-Alert` which works Ok and has pretty much the same readability.

Also, IIRC, this behavior was apparent if a message was entered manually into Meshchat or curl'ed in from an external app.

This behavior might also be applicable to Issues 11 and 21.

BTW, my weather alert code has been fixed for some time now. To see it in action, go to `Meshchat-WX`. The url should be in many Mesh Status pages.
